### ISSUE TITLE
Working with git 

### ISSUE THREAD

**Issue by CleanMachine1** [CleanMachine1/DollarSkip#19]
Running commands such as

```
$ git commit -m "test 123" --allow-empty
error: pathspec '123' did not match any file(s) known to git
```

Gives an error

**Comment by Itai-Nelken** [comment]
#20 might fix this as the command is stored in an array like this:
```
_________________________________________________
| git | commit | -m | "test 123" | --allow-empty |
 ------------------------------------------------

```

**Comment by Itai-Nelken** [comment]
it does!
```
itai@pinebookpro:~$ $ git commit -m "test 123" --allow-empty
fatal: not a git repository (or any of the parent directories): .git
itai@pinebookpro:~$
```
(with the fixes I'll push to the branch now)

**Comment by CleanMachine1** [comment]
I haven't seen the PR yet, however did you try it in a repo, only occurs when it is in a repo

**Comment by Itai-Nelken** [comment]
I created a test repo (with `mkdir test; cd test; git init) and made a test commit and it works!
```
itai@pinebookpro:~/Desktop/test$ git commit -m "test 123" --allow-empty
[master (root-commit) ec5a102] test 123
itai@pinebookpro:~/Desktop/test$ git log
commit ec5a1022c72cb3dc3634fd1248946d10a3d4bcde (HEAD -> master)
Author: Itai Nelken <itainelken@gmail.com>
Date:   Sun Jul 11 22:23:14 2021 +0300

    test 123
itai@pinebookpro:~/Desktop/test$
```

**Comment by Itai-Nelken** [comment]
It doesn't work... I forgot the $ when testing...

**Comment by Itai-Nelken** [comment]
I found the cause of the problem (using execvp()) (ignore the segfault):
```
itai@pinebookpro:~/Desktop/test$ $ git commit -m '1 2' --allow-empty
git
commit
-m
1
2
--allow-empty
Segmentation fault (core dumped)
itai@pinebookpro:~/Desktop/test$ $ git commit -m '1 2' --allow-empty
error: pathspec '2' did not match any file(s) known to git
itai@pinebookpro:~/Desktop/test$ 
```
the '1 2' is threated as 2 different arguments.

**Comment by Itai-Nelken** [comment]
The problem is that what DollarSkip gets from this input: `$ git commit -m "test 123" --allow-empty` is `git commit -m test 123 --allow-empty` (note the missing double quotes). that is bash's fault, when using the shell I wrote that doesn't handle quotes and double quotes, DollarSkip gets the correct string (`git commit -m "test 123" --allow-empty`) and we can process it to make the characters between the double quotes be a single argument.

TL;DR the problem is that we need to modify the shell to fix this or - I don't know if it is possible - we need to change shell settings to leave quotes and double quotes alone when passing arguments to DollarSkip.

**Comment by Itai-Nelken** [comment]
This might work: https://stackoverflow.com/questions/10835933/preserve-quotes-in-bash-arguments

We can use sprintf() to "print" the arguments into a variable.

**Comment by Itai-Nelken** [comment]
Didn't work :(

**Comment by CleanMachine1** [comment]
I have now completed the go version after actually learning some basic concepts in go and getting a grip on it.

The current version can be seen [here](https://github.com/CleanMachine1/DollarSkip/blob/dollarskip-go/dollarskip.go) and I plan to merge it very soon.
